beetsplug/parentwork.py

```diff
@@ -0,0 +1,197 @@
+                                              includes=["artist-rels"])
+    return work_info, work_date
+
+
+class ParentWorkPlugin(BeetsPlugin):
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+
+        self.config.add({
+            'auto': False,
+            'force': False,
+        })
+
+        self._command = ui.Subcommand(
+            'parentwork',
+            help=u'Fetches parent works, composers and dates')
+
+        self._command.parser.add_option(
+            u'-f', u'--force', dest='force',
+            action='store_true', default=None,
+            help=u'Re-fetches all parent works')
+
+        if self.config['auto']:
+            self.import_stages = [self.imported]
+
+    def commands(self):
+
+        def func(lib, opts, args):
+            self.config.set_args(opts)
+            force_parent = self.config['force'].get(bool)
+            write = ui.should_write()
+
+            for item in lib.items(ui.decargs(args)):
+                self.find_work(item, force_parent)
+                item.store()
+                if write:
+                    item.try_write()
+
+        self._command.func = func
+        return [self._command]
+
+    def imported(self, session, task):
+        """Import hook for fetching parent works automatically.
+        """
+        force_parent = self.config['force'].get(bool)
+
+        for item in task.imported_items():
+            self.find_work(item, force_parent)
+            item.store()
+
+    def get_info(self, item, work_info):
+        """Given the parentwork info dict, this function fetches
+        parent_composer, parent_composer_sort, parentwork,
+        parentwork_disambig, mb_workid and composer_ids"""
+
+        parent_composer = []
+        parent_composer_sort = []
+
+        composer_exists = False
+        if 'artist-relation-list' in work_info['work']:
+            for artist in work_info['work']['artist-relation-list']:
+                if artist['type'] == 'composer':
+                    parent_composer.append(artist['artist']['name'])
+                    parent_composer_sort.append(artist['artist']['sort-name'])
+        if not composer_exists:
+            self._log.info(item.artist + ' - ' + item.title)
+            self._log.debug(
+                "no composer, add one at https://musicbrainz.org/work/" +
+                work_info['work']['id'])
+        parentwork = work_info['work']['title']
+        mb_parentworkid = work_info['work']['id']
+        if 'disambiguation' in work_info['work']:
+            parentwork_disambig = work_info['work']['disambiguation']
+            return [parentwork, mb_parentworkid, parent_composer,
+                    parent_composer_sort, parentwork_disambig]
+        else:
+            return [parentwork, mb_parentworkid, parent_composer,
+                    parent_composer_sort, None]
+
+    def find_work(self, item, force):
+
+        recording_id = item.mb_trackid
```

Comment:
I think it would be OK (and a little clearer) to just use `item.mb_trackid` instead of putting it in a variable here.